src/TraceEvent/Parsers/ClrTraceEventParser.cs

```diff
@@ -191,7 +193,44 @@ public enum Keywords : long
             /// </summary>
             GCHeapSnapshot = GC | GCHeapCollect | GCHeapDump | GCHeapAndTypeNames | Type,
         };
-        public ClrTraceEventParser(TraceEventSource source) : base(source) { }
+        public ClrTraceEventParser(TraceEventSource source) : base(source)
```

Comment:
At one time the ClrTraceEventParser was generated using TraceParserGen.   However TraceParserGen does not support a bunch of cases, including arrays of structs, and other things that were added later on by hand.   It also does not support parsers with state.   You could imagine doing this, but it is relatively rare and has never been done.   When we add new events to ClrTraceEventParser we run the tool and then bascally hand merge the new parts into the existing code.